### ISSUE TITLE
Incompatibility with 'Rails 4.2.0' and 'Responders 2.0.2'.

### ISSUE THREAD

**Issue by esbanarango** [rails-api/rails-api#179]
Hello,

I'm trying to upgrade my app to `Rails 4.2.0`. My version of rails-api is `0.3.1`. I'm using `respond_to` and `respond_with` features which are now extracted to the `responders` gem. I already added the responders gem to my gemfile, but I'm still having the `NoMethodError`.

```
.../gems/actionpack-4.2.0/lib/action_controller/metal/mime_responds.rb:10:in `respond_to': The controller-level `respond_to' feature has been extracted to the `responders` gem. Add it to your Gemfile to continue using this feature: (NoMethodError)
  gem 'responders', '~> 2.0'

```

This is my `application_controller.rb`:

``` ruby

class ApplicationController < ActionController::API

  include ActionController::MimeResponds
  include ActionController::ImplicitRender

  respond_to :json
end


```

**Comment by runwaldarshu** [comment]
I have responders gem in my gemspec file and with that I get the same error.
If I add it in gemfile it works but I really want it to come from gemspec so that applications using that gem do not have to include it in their gemfile

**Comment by abrisse** [comment]
It won't work if you need the instance-level `respond_to` method which is only defined in MimeResponds module (see https://github.com/rails/rails/issues/21469)

**Comment by yogeshjain999** [comment]
Agree with @iRonin about not removing `ActionController::MimeResponds`. Also, remember to include it after `ActionController::RespondWith`.

**Comment by sunnydgoswami** [comment]
https://stackoverflow.com/a/46554377/8518019

in gemfile add - gem 'responder', '~> 2.0'
then run rails generate responder:install

then run your rails generate scaffold *abcdefgh*

If you have db issues I suggest running db:drop, db:create and then doing a rake db:migrate. I faced a migration issue so I just dropped my database and created a new one and a new migration. 

This solved the issues for me.
